std/experimental/allocator/building_blocks/affix_allocator.d

```diff
@@ -124,17 +124,19 @@ struct AffixAllocator(Allocator, Prefix, Suffix = void)
             if (!bytes) return null;
             auto result = parent.allocate(actualAllocationSize(bytes));
             if (result is null) return null;
-            static if (stateSize!Prefix)
-            {
-                assert(result.ptr.alignedAt(Prefix.alignof));
-                emplace!Prefix(cast(Prefix*) result.ptr);
-            }
-            static if (stateSize!Suffix)
-            {
-                auto suffixP = result.ptr + result.length - Suffix.sizeof;
-                assert(suffixP.alignedAt(Suffix.alignof));
-                emplace!Suffix(cast(Suffix*)(suffixP));
-            }
+            () @trusted {
+                static if (stateSize!Prefix)
+                {
+                    assert(result.ptr.alignedAt(Prefix.alignof));
+                    emplace!Prefix(cast(Prefix*) result.ptr);
```

Comment:
This assumes that it's safe to write over `result`. As far as I see, you can't assume that. Every array type implicitly converts to `void[]`, so it's possible that `parent.allocate` returns memory that's already used elsewhere, differently typed.

Same with `Suffix` below.

Reply:
Isn't it safe to assume that `parent.allocate` will return fresh memory? Imho, `allocate` should always return fresh memory.

Reply:
> Isn't it safe to assume that parent.allocate will return fresh memory?

No. `parent` is provided by the user. You can't assume anything about it beyond what is guaranteed/required by the language.

It might be possible to check if the return value is unique. Then you can rely on that. But you have to check.

Reply:
This is problematic. We may need to replace the result of allocate with `ubyte[]` everywhere, or give up on making allocate safe for this allocator.